simulator/func_sim/alu.h

```diff
@@ -240,6 +240,18 @@ struct ALU
     // Bit manipulations
     template<typename I> static void sbext( I* instr) { instr->v_dst = 1 & ( instr->v_src1 >> shamt_v_src2<typename I::RegisterUInt>( instr)); }
 
+    template<typename I, typename T>
+    static void clmul( I* instr)
+    {
+        instr->v_dst = 0;
+        std::size_t max = bitwidth<T> - count_leading_zeroes<T>( instr->v_src2);
+        for ( std::size_t index = 0; index < max; index++)
+        {
```

Comment:
Please be consistent, either remove brackets around loop body (preferable) or add brackets around 'then' (less preferable).